docs/editing-components.md

```diff
@@ -0,0 +1,376 @@
+# Creating and Editing Wagtail Components
+
+cfgov-refresh implements a number of components
+that editors can choose from when building a page,
+for example: Heroes, Expandable Groups, or Info Unit Groups.
+In Wagtail parlance, these are called
+["StreamField blocks"](https://docs.wagtail.io/en/v1.13.4/topics/streamfield.html)*
+(or just "blocks").
+We sometimes also refer to them as "modules", because we think that
+the terms "component" and "module" may be more obvious to non-developers.
+This page will use the terms somewhat interchangeably.
+One other important thing to note before we begin:
+blocks can be nested within other blocks.
+
+_* If you're going to be doing anything more than making minor updates to
+   existing components, this is highly recommended reading._
+
+
+## Table of contents
+
+1. [The parts of a Wagtail block](#the-parts-of-a-wagtail-block)
+   1. [The Python class](#the-python-class)
+   1. [Adding it to a StreamField](#adding-it-to-a-streamfield)
+   1. [The HTML template](#the-html-template)
+   1. [Adding some style](#adding-some-style)
+   1. [Adding some JavaScript](#adding-some-javascript)
+1. [How-to guides](#how-to-guides)
+   1. [Adding, editing, or removing fields in the Wagtail editor](#adding-editing-or-removing-fields)
+   1. [Creating migrations for StreamField blocks](#creating-migrations-for-streamfield-blocks)
+
+
+## The parts of a Wagtail block
+
+Blocks are implemented via several different bits of code:
+
+1. [Defining a block's fields and other properties in a Python class](#the-python-class)
+1. [Adding the class to a page's StreamField block options](#adding-it-to-a-streamfield)
+1. [Creating an HTML template for rendering the block on a page](#the-html-template)
+1. [(Optionally) adding some Less for styling the block](#adding-some-style)
+1. [(Optionally) adding some JavaScript for adding advanced behavior](#adding-some-javascript)
+
+Before you dive in further,
+[check out the Atomic Structure page](../atomic-structure/)
+and familiarize yourself with our basic concepts
+of atoms, molecules, and organisms.
+
+
+### The Python class
+
+A component's fields and other properties are defined in a Python class file
+corresponding to the level of the component.
+
+These files—`atoms.py`, `molecules.py`, and `organisms.py`—are located in
+[`cfgov/v1/atomic_elements`](https://github.com/cfpb/cfgov-refresh/tree/master/cfgov/v1/atomic_elements).
```

Comment:
We have [other apps that use `blocks.py`](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/data_research/blocks.py), which is more in line with Wagtail convention. We should make a distinction between blocks that need to be available to every part of the CMS (maybe to our general pages, Learn Page, etc) and those that are specific to app contexts (like the Data & Research blocks linked to above, or say, if TDP needed to create a custom block for their Activity Pages).

Reply:
Addressed in a47039c